File: Code/Python/Challenge006.py
import helper
dir=".\\Data\\006"
helper.ensuredir(dir)

import urllib.request
channel="http://www.pythonchallenge.com/pc/def/channel.zip"
(filename, headers)=urllib.request.urlretrieve(channel, dir+"\\channel.zip")

import zipfile
zip=zipfile.ZipFile(filename)

import re
pattern=re.compile("Next nothing is (\\d+)")

entryname="90052"
outstr=[]
while True:
    name=entryname+".txt"
    cnt=zip.read(name).decode('utf-8')
    outstr.append(zip.getinfo(name).comment.decode('utf-8'))
    try:
        entryname=pattern.search(cnt).group(1)
    except:
        print("last: %s, cnt: %s" %(entryname, cnt))
        break

print(''.join(outstr))

File: docs/Code/Go/Challenge006.go
package main

import (
	"fmt"
	"archive/zip"
	"io/ioutil"
	"regexp"
)

func (c *Challenge) Challenge006() {
	path:=".\\Data\\006"
	EnsureDir(path)

	filename:=path+"\\channel.zip"
	channel:="http://www.pythonchallenge.com/pc/def/channel.zip"
	Download(channel,filename)

	rc,err:=zip.OpenReader(filename)
	if err != nil {
		fmt.Printf("open zip failed![%v]\n", err)
	}

	defer rc.Close()

	dic:=map [string] *zip.File {}
	for _,file := range rc.File {
		dic[file.Name]=file
	}

	outstr:=""
	pattern:=regexp.MustCompile("Next nothing is (\\d+)")
	name:="90052"
	for true {
		file,exists:=dic[name+".txt"]
		if exists {
			outstr+=file.Comment
			rc,err:=file.Open()
			if err == nil {
				defer rc.Close()
				bytes,err:=ioutil.ReadAll(rc)
				if err == nil {
					cnt:=string(bytes)
					matches:=pattern.FindStringSubmatch(cnt)
					if len(matches) != 2 {
						fmt.Printf("last: %s, cnt: %s\n", name, cnt)
						break
					} else {
						name=matches[1]
					}
				}
			}
		}
	}

	fmt.Println(outstr)
}